docs/lesson-plans/week-5.md

```diff
@@ -0,0 +1,639 @@
+(like dependencies). It lets you create fake versions of objects which you have complete control over, and it even lets
+you verify things about how it was called.
+
+If you're interested in learning more about Mockito, [here is the 
+documentation](http://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html). It even lets you do some 
+really cool things like `spy()`, where you can pass in an existing object, have it do _most_ of the things it usually 
+does, and then just override/verify/mock what you want!
+
+#### Mocking an Object and Overriding Returns
+
+To mock an object, all you have to do is call (the static method) `mock(Thing.class)`, which will return a fake
+instance of that `Thing` class/interface. By default, it will return null 
+for every non-void method you call, but you can override that using `thenReturn`, `thenThrow`, `then` (`thenAnswer`, 
+allows you to do custom stuff), or `doReturn`, `doThrow`, `doAnswer`, `doNothing`, `doCallRealMethod` in the case of 
+void returns. You can also chain them together to have it return/throw/do things one after the other in the order 
+they're called.
+
+>An 'instance' of an object is a constructed value of the object created by its constructor. So for the class `Thing`, 
+>it's the difference between referring to `Thing.staticMethod()` as a static method and `new Thing().instanceMethod()`.
+
+Here are examples of the `then...` and `do...` styles of mocking:
+```java 
+MyCustomObject obj = mock(MyCustomObject.class); // can be done to interfaces too!
+
+when(obj.myCustomMethod()).thenReturn("Hello World!");
+
+String res = obj.myCustomMethod() // Returns "Hello World!");
+
+assertEquals("Hello World!", res);
+```
+
+```java 
+MyCustomObject obj = mock(MyCustomObject.class); // can be done to interfaces too!
+
+doReturn("Hello World!").when(obj).myCustomMethod());
+
+String res = obj.myCustomMethod() // Returns "Hello World!");
+
+assertEquals("Hello World!", res);
+```
+
+If you're interested in having the thing you want to do be done only when certain parameters are entered, then you can 
+do that with argument matchers. You can do things like `any()` for anything, `anyInt()` for any integer, 
+`any(MyClass.class)` for that specific class, or even just enter your own number/string for specific entries.
+The result you specify will only be returned/done when that matcher is satisfied. 
+
+>Note: if there's another condition you need satisfied, you can also create your own matcher if the default ones aren't 
+>enough (I've never done that though, the default ones are usually good enough).
+
+Example:
+```java 
+when(myMock.do(anyInt()).thenReturn(5);
+
+myMock.do(5); // returns 5
+myMock.do(-1); // returns 5
+myMock.do(anyInt()); // you can't do this though, it only works in the "when" or "do..." methods
+// This is because you're actually calling the method (or the mocked version), so you need to pass in a real value
+
+when(myMock.other(any(Other.class)).thenReturn(5);
```

Comment:
Pick a better method name than `other` and call the class OtherClass instead of just Other